test/Amazon.Lambda.Tools.Integ.Tests/DeployProjectTests.cs

```diff
@@ -278,5 +278,153 @@ public async Task PackageFunctionAsLocalImageThenDeployWithDifferentECRTag()
+
+                // Redeploy changing the ephemeral size and clearning environment variables.
+                command = new DeployFunctionCommand(toolLogger, fullPath, new string[0]);
+                command.FunctionName = functionName;
+                command.Role = await TestHelper.GetTestRoleArnAsync();
+                command.Runtime = "dotnet6";
+                command.EphemeralStorageSize = 800;
+                command.EnvironmentVariables = new System.Collections.Generic.Dictionary<string, string> ();
+                command.DisableInteractive = true;
+
+                created = await command.ExecuteAsync();
+                Assert.True(created);
+
+                getConfigResponse = await command.LambdaClient.GetFunctionConfigurationAsync(functionName);
+                Assert.Equal(800, getConfigResponse.EphemeralStorage.Size);
+                Assert.Null(getConfigResponse.Environment);
+            }
+            finally
+            {
+                try
+                {
+                    await command.LambdaClient.DeleteFunctionAsync(functionName);
+                }
+                catch { }
+            }
+        }
+
+        [Fact]
+        public async Task DeployWithFunctionUrl()
+        {
+            using var httpClient = new HttpClient();
+            var toolLogger = new TestToolLogger(_testOutputHelper);
+
+            var fullPath = Path.GetFullPath(Path.GetDirectoryName(this.GetType().GetTypeInfo().Assembly.Location) + "../../../../../../testapps/TestHttpFunction");
+            var functionName = "DeployWithFunctionUrl-" + DateTime.Now.Ticks;
+
+            async Task TestFunctionUrl(string functionUrl, bool expectSuccess)
+            {
+                var httpResponse = await httpClient.GetAsync(functionUrl);
+                Assert.Equal(expectSuccess, httpResponse.IsSuccessStatusCode);
+                httpResponse.Dispose();
+            }
+
+            async Task TestPublicPermissionStatement(IAmazonLambda lambdaClient, bool expectExist)
+            {
+                if(expectExist)
+                {
+                    var policy = (await lambdaClient.GetPolicyAsync(new GetPolicyRequest { FunctionName = functionName })).Policy;
+                    Assert.Contains(LambdaConstants.FUNCTION_URL_PUBLIC_PERMISSION_STATEMENT_ID, policy);
+                }
+                else
+                {
+                    try
+                    {
+                        var policy = (await lambdaClient.GetPolicyAsync(new GetPolicyRequest { FunctionName = functionName })).Policy;
+                        Assert.DoesNotContain(LambdaConstants.FUNCTION_URL_PUBLIC_PERMISSION_STATEMENT_ID, policy);
+                    }
+                    catch (ResourceNotFoundException)
+                    {
+                        // If the last statement is deleted from the policy then a ResourceNotFoundException is thrown which is also proof the statement id has been removed.
```

Comment:
Is your test guaranteed to always throw? You could replace this try catch with something like `Assert.ThrowsAsync<ResourceNotFoundException>( fnCall )`.

Reply:
What I'm asserting is the permission statement has been removed. If being removed meant the last statement was removed then you get the `ResourceNotFoundException` exception. In my test currently that is always the case but I want to leave it flexible in case things change in the future where the public statement we added was removed but others might exist.